core/modules/loader/appClientNum.cc

```diff
@@ -0,0 +1,277 @@
+                ++successCount;
+            } else {
+                ++failedCount;
+                LOGS(_log, LOG_LVL_WARN, "insert failed " << *kPtr);
+            }
+            iter = kList.erase(iter);
+        } else {
+            ++iter;
+        }
+    }
+    return kList.empty();
+}
+
+
+/// Get a repeatable value for the chunk and subchunk numbers. It's arbitrary for
+/// the test as there just needs to be some check that what was written in for
+/// the key is the same as what was read
+int calcChunkFrom(uint64_t j) {
+    return j%10000;
+}
+int calcSubchunkFrom(uint64_t j) {
+    return j%100;
+}
+
+
+KeyInfoData::Ptr clientAdd(CentralClient& central, uint64_t j) {
+    CompositeKey cKey(j);
+    int chunk = calcChunkFrom(j);
+    int subchunk = calcSubchunkFrom(j);
+    return central.keyInsertReq(cKey, chunk, subchunk);
+}
+
+
+/// @return true if the list is empty.
+bool keyLookupListClean(KeyInfoDataList& kList, int& successCount, int& failedCount) {
+    for(auto iter=kList.begin(); iter != kList.end();) {
+        if (*iter == nullptr || (*iter)->isFinished()) {
+            KeyInfoData::Ptr const& kPtr = *iter;
+            if (kPtr->success) {
+                // check the values
+                uint64_t j = kPtr->key.kInt;
+                // expected chunk and subchunk values.
+                int expChunk = calcChunkFrom(j);
+                int expSubchunk = calcSubchunkFrom(j);
+                if (kPtr->chunk == expChunk && kPtr->subchunk == expSubchunk) {
+                    ++successCount;
+                } else {
+                    ++failedCount;
+                    LOGS(_log, LOG_LVL_WARN, "lookup failed, bad values, expected c=" << expChunk <<
+                         " sc=" << expSubchunk << " found=" << *kPtr);
+                }
+            } else {
+                ++failedCount;
+                LOGS(_log, LOG_LVL_WARN, "lookup failed " << *kPtr);
+            }
+            iter = kList.erase(iter);
+        } else {
+            ++iter;
+        }
+    }
+    return kList.empty();
+}
+
+
+KeyInfoData::Ptr clientAddLookup(CentralClient& central, uint64_t j) {
+    CompositeKey cKey(j);
+    return central.keyLookupReq(cKey);
+}
+
+
+int main(int argc, char* argv[]) {
+    std::string cCfgFile("core/modules/loader/config/client1.cnf");
+    if (argc < 3) {
+        LOGS(_log, LOG_LVL_ERROR, "usage: appClientNum <startingNumber> <endingNumber> <optional config file name>");
+        return 1;
+    }
+    uint64_t numStart = boost::lexical_cast<uint64_t>(argv[1]);
+    uint64_t numEnd   = boost::lexical_cast<uint64_t>(argv[2]);
+    if (argc > 3) {
+        cCfgFile = argv[3];
+    }
+    LOGS(_log, LOG_LVL_INFO, "start=" << numStart << " end=" << numEnd << " cCfg=" << cCfgFile);
+    if (numEnd == 0) {
+        LOGS(_log, LOG_LVL_ERROR, "end cannot equal 0");
+        return 1;
+    }
+
+    std::string ourHost;
+    {
+        char hName[300];
+        if (gethostname(hName, sizeof(hName)) < 0) {
```

Comment:
I would rather use BOOST ASIO since you already using it:
```
std::string const name = boost::asio::ip::host_name();
```
It's here:
```
namespace boost {
namespace asio {
namespace ip {

/// Get the current host name.
BOOST_ASIO_DECL std::string host_name();

/// Get the current host name.
BOOST_ASIO_DECL std::string host_name(boost::system::error_code& ec);

} // namespace ip
} // namespace asio
} // namespace boost
```